### PR TITLE
human output suppression and emit_log_level

### LINKED ISSUE #26: allow suppressing err
Essentially, I'd like to purposely suppress messages like:
` [fcrdns] queryA ESERVFAIL jaana24.stonemarketing.com.br` which are triggered by `connection.results.add(plugin, {err: err2[e].message})`

I don't think error level is appropriate for irrelevant things. 

Or, better yet, just allow specifying the level, like `results.add(plugin, {err: blah, log_level: 'info'})`

What do you think?

### PR DESCRIPTION
**Pull request by celesteking**
Fixes #26

### CODE REVIEW

**Comment by celesteking** [comment]
Well, https://github.com/haraka/haraka-results/pull/11#issuecomment-381357165 says `human` param should actually be saved into results and not displayed, which is very confusing. I though it's for displaying in logs and not for later processing (why would you hard-code something instead of making a string ID that would later be translated/expanded to user's language?)

File: index.js [hunk 1]
```diff
@@ -94,6 +94,7 @@ class ResultStore {
 
         // anything else is an arbitrary key/val to store
         for (const key in obj) {
+            if (key.match(/^human(_html)?$/)) continue;
```
Comment (msimerson) [inline index.js]: this is redundant. `human` and `human_html` are in `all_opts` and will get skipped by the very next line.

File: index.js [hunk 2]
```diff
@@ -133,6 +134,7 @@ class ResultStore {
         this.redis_publish(name, obj);
 
         for (const key in obj) {
+            if (key.match(/^human(_html)?$/)) continue;
```
Comment (msimerson) [inline index.js]: same.

File: index.js [hunk 3]
```diff
@@ -221,15 +223,12 @@ class ResultStore {
...
         if (obj.err) {
             // Handle error objects by logging the message
-            if (util.isError(obj.err)) {
-                this.conn.logerror(plugin, obj.err.message);
-            }
-            else {
-                this.conn.logerror(plugin, obj.err);
-            }
+            const level = obj.emit_log_level ? obj.emit_log_level : 'error';
+            const logobj = util.isError(obj.err) ? obj.err.message : obj.err;
```
Comment (msimerson) [inline index.js]: The name `logobj` implies that the contents are an object and that's not the case. Lets call this 'logmsg' or even just 'msg', at your discretion.

File: index.js [hunk 4]
```diff
@@ -221,15 +223,12 @@ class ResultStore {
...
+        if (obj.emit  && result.human.trim().length > 0) this.conn.loginfo(plugin, result.human);  // by request
         if (obj.err) {
             // Handle error objects by logging the message
-            if (util.isError(obj.err)) {
-                this.conn.logerror(plugin, obj.err.message);
-            }
-            else {
-                this.conn.logerror(plugin, obj.err);
-            }
+            const level = obj.emit_log_level ? obj.emit_log_level : 'error';
```
Comment (msimerson) [inline index.js]: This name of `emit_log_level` is misleading. When emit is specified, the result is logged at loginfo, as statically defined a few lines above. If one were to specify an `emit_log_level`, wouldn't they expect the level of the `emit: true` to change? 

This should be named differently, perhaps as `err_log_level`.

**Comment by msimerson** [comment]
abandoned.

**Pull request closed without merging**